### PR TITLE
Prepare release 0.101.1

### PR DESCRIPTION
**Pull request by alejoe91**
- [x] Update pyproject toml (remove git installs)
- [x] Update `DEV_MODE` in init
- [x] Add release notes
- [x] Update `whatisnew`
- [x] Tag KS4 4.0.16 Docker image as `latest`

### CODE REVIEW

File: doc/releases/0.101.1.rst [hunk 1]
```diff
@@ -0,0 +1,113 @@
+.. _release0.101.0:
+
+SpikeInterface 0.101.0 release notes
```
Comment (zm711) [inline doc/releases/0.101.1.rst:3]:
```suggestion
SpikeInterface 0.101.1 release notes
```
right?

File: doc/releases/0.101.1.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
+.. _release0.101.0:
```
Comment (zm711) [inline doc/releases/0.101.1.rst:1]:
```suggestion
.. _release0.101.1:
```

**Comment by zm711** [comment]
Wow development is accelerating. Just this morning we were working on 0.101.1. :)

**Comment by alejoe91** [comment]
Apparently #3369 is a breaking change, so we should reflect it? :)

**Comment by zm711** [comment]
That's fair :)

**Comment by alejoe91** [comment]
@zm711 we decided to refactor https://github.com/SpikeInterface/spikeinterface/pull/3369 so it's back-compatible, by adding a `from_amplitude` sparsity mode. Then we're back to 0.101.1 :)

**Review by samuelgarcia — APPROVED** [review]

File: doc/releases/0.101.1.rst [hunk 3]
```diff
@@ -0,0 +1,143 @@
+.. _release0.101.1:
+
+SpikeInterface 0.101.1 release notes
+------------------------------------
+
+13th September 2024
```
Comment (zm711) [inline doc/releases/0.101.1.rst:6]:
```suggestion
16th September 2024
```
or 17th if you think you'll take more time :)

File: doc/whatisnew.rst [hunk 4]
```diff
@@ -43,6 +44,16 @@ Release notes
   releases/0.9.1.rst
 
 
+Version 0.101.1
+===============
+
+* Enabled support for consolidated Zarr-backend for `SortingAnalyzer`, including cloud support (#3314, #3318, #3349, #3351)
+* Improved support for Kilosort4 **ONLY VERSIONS >= 4.0.16** (#3339, #3276)
+* Skip recomputation of quality and template metrics if already computed (#3292)
+* Improved `estimate_sparsity` with new `amplitude` method and and deprecated `from_ptp` option (#3369)
```
Comment (zm711) [inline doc/whatisnew.rst:53]:
```suggestion
* Improved `estimate_sparsity` with new `amplitude` method and deprecated `from_ptp` option (#3369)
```

File: doc/releases/0.101.1.rst [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+* Test IBL skip when the setting up the one client fails (#3289)
+* Add kilosort4 wrapper tests (#3085)
+
+Contributors:
+
+* @Djoels
+* @JoeZiminski
+* @JuanPimientoCaicedo
+* @alejoe91
+* @app/pre-commit-ci
```
Comment (zm711) [inline doc/releases/0.101.1.rst:132]:
```suggestion
```
I assume we don't care about saying pre-commit-ci committed :)

**Review by zm711 — COMMENTED** [review]
A couple tiny touch-ups.

File: doc/releases/0.101.1.rst [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+------------------------------------
+
+13th September 2024
+
+Main changes:
+
+* Enabled support for consolidated Zarr-backend for `SortingAnalyzer`, including cloud support (#3314, #3318, #3349, #3351)
+* Improved support for Kilosort4 **ONLY VERSIONS >= 4.0.16** (#3339, #3276)
+* Skip recomputation of quality and template metrics if already computed (#3292)
+* Improved `estimate_sparsity` with new `amplitude` method and and deprecated `from_ptp` option (#3369)
```
Comment (alejoe91) [inline doc/releases/0.101.1.rst:13]:
```suggestion
* Improved `estimate_sparsity` with new `amplitude` method and deprecated `from_ptp` option (#3369)
```

**Comment by zm711** [comment]
Looks like there are Kachery errors now. This isn't related to your other PR is it?

**Comment by alejoe91** [comment]
> Looks like there are Kachery errors now. This isn't related to your other PR is it?

It is. I had changed secrets but reverted now :) We'll deal with kachery later. Tests should pass now (hopefully!)

**Pull request merged**